include/hc.hpp

```diff
@@ -755,10 +756,52 @@ class accelerator
      * @return An opaque handle of the underlying HSA agent, if the accelerator
      *         is based on HSA.  NULL otherwise.
      */
-    void* get_hsa_agent() {
+    void* get_hsa_agent() const {
         return pDev->getHSAAgent();
     }
 
+    /**
+     * Check if @p other is peer of this accelerator.
+     *
+     * @return true if other can access this accelerator's device memory pool or false if not
+     */
+    bool is_peer(const accelerator& other) const {
+      hsa_amd_memory_pool_t* self_pool = static_cast<hsa_amd_memory_pool_t*>(get_hsa_am_region());  
```

Comment:
The implementation of hc::accelerator::is_peer() directly uses AMD HSA extension API. It breaks separation of interface and implementation we have so far, and would make supporting other targets (x86 CPU / OpenCL) rather hard. Please try define a virtual member function in KalmarDevice, and have the HSA implementation placed in mcwamp_hsa.cpp .